README.md

```diff
@@ -1,5 +1,87 @@
 
-# Contributing
+# Microsoft Authentication Extensions for Python
+
+The Microsoft Authentication Extensions for Python offers secure mechanisms for public client applications to perform cross-platform token cache serialization and persistence. It gives additional support to the [Microsoft Authentication Library for Python (MSAL)](https://github.com/AzureAD/microsoft-authentication-library-for-python). 
+
+MSAL Python supports an in-memory cache by default and provides the [SerializableTokenCache](https://msal-python.readthedocs.io/en/latest/#msal.SerializableTokenCache) to perform cache serialization. You can read more about this in the MSAL Python [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-python-token-cache-serialization). Developers are required to implement their own cache persistance across multiple platforms and Microsoft Authentication Extensions makes this simpler. The extensions library supports persistence of the cache to disk in encrypted form where the platform supports this.
+
+This is available for Windows, Mac and Linux. 
+- Windows - [DPAPI](https://docs.microsoft.com/en-us/dotnet/standard/security/how-to-use-data-protection) is used for encryption.
+- MAC - The MAC KeyChain is used and encryption is built in.
+- Linux - [LibSecret](https://wiki.gnome.org/Projects/Libsecret) is used for encryption. Any place where libsecret is not available encryption is not supported.
+
+> Note: This library provides cache persistance support for Public client applications such as Desktop apps only. Web applications are recommended to persist the cache in memory or in Redis. Take a look at this [webapp sample](https://github.com/Azure-Samples/ms-identity-python-webapp).
+
+## Installation
+
+You can find Microsoft Authentication Extensions for Python on [Pypi](https://pypi.org/project/msal-extensions/).
+1. If you haven't already, [install and/or upgrade the pip](https://pip.pypa.io/en/stable/installing/)
+   of your Python environment to a recent version. We tested with pip 18.1.
+2. Run `pip install msal-extensions`.
+
+## Versions
+
+This library follows [Semantic Versioning](http://semver.org/).
+
+You can find the changes for each version under
+[Releases](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/releases).
+
+## Usage
+
+The Microsoft Authentication Extensions library provides the `PersistedTokenCache` which accepts a platform-dependent persistence instance. This token cache can then be used to instantiate the `PublicClientApplication` in MSAL Python. 
+
+The token cache includes a file lock, and auto-reload behavior under the hood.
+
+
+
+Here is an example of this pattern for multiple platforms:

```

Comment:
Perhaps also provide a link to the real sample, in case that sample would receive some new enhancement or bugfix in future, while we might not remember to update the code snippet inside this README?

UPDATED: Oh, after finishing reading this entire section, I realize that you did provide a link to that sample at the end. Thanks! But would it still be better to provide that link up front?

```suggestion
Here is an example of this pattern for multiple platforms
(and its latest implementation is available
[here](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/blob/dev/sample/persistence_sample.py)
):

```